e2e/suites/management/autoscaling_test.go

```diff
@@ -0,0 +1,258 @@
+								"curl --request PUT $ROOMS_API_ADDRESS/scheduler/$MAESTRO_SCHEDULER_NAME/rooms/$MAESTRO_ROOM_ID/ping --data-raw '{\"status\": \"ready\",\"timestamp\": \"12312312313\"}' && " +
+								"sleep 5 ; done",
+							},
+						},
+					}},
+				Autoscaling: &maestroApiV1.OptionalAutoscaling{
+					Enabled: &enabled,
+					Min:     &min,
+					Max:     &max,
+					Policy:  &policy,
+				},
+			}
+			patchSchedulerResponse := &maestroApiV1.PatchSchedulerResponse{}
+			err = managementApiClient.Do("PATCH", fmt.Sprintf("/schedulers/%s", scheduler.Name), patchSchedulerRequest, patchSchedulerResponse)
+			require.NoError(t, err)
+
+			require.Eventually(t, func() bool {
+				schedulerInfoRequest := &maestroApiV1.GetSchedulersInfoRequest{}
+				schedulerInfoResponse := &maestroApiV1.GetSchedulersInfoResponse{}
+				err = managementApiClient.Do("GET", fmt.Sprintf("/schedulers/info?game=%s", scheduler.Game), schedulerInfoRequest, schedulerInfoResponse)
+				require.NoError(t, err)
+				require.NotEmpty(t, schedulerInfoResponse.GetSchedulers())
+				schedulerInfo := getSchedulerInfo(scheduler.Name, schedulerInfoResponse.GetSchedulers())
+
+				return schedulerInfo.GetRoomsReady() == int32(10)
+			}, time.Minute*2, time.Second*1, "Scheduler should initially have 10 ready rooms")
+
+			// Now, enable autoscaling
+			enabled = true
+
+			// Rooms will keep getting occupied to force a full scale up
+			patchSchedulerRequest = &maestroApiV1.PatchSchedulerRequest{
+				Autoscaling: &maestroApiV1.OptionalAutoscaling{
+					Enabled: &enabled,
+				}}
+			patchSchedulerResponse = &maestroApiV1.PatchSchedulerResponse{}
+			err = managementApiClient.Do("PATCH", fmt.Sprintf("/schedulers/%s", scheduler.Name), patchSchedulerRequest, patchSchedulerResponse)
+			require.NoError(t, err)
+
+			// Expect rooms to full scale down trying to maintain the 50% ready target since no room will get occupied
+
+			// 1 ready and 0 occupied
+			requireEventualState(t, 1, 0, scheduler.Name, managementApiClient)
+		})
+
+	})
+}
+
+func requireEventualState(t *testing.T, numberOfReadyRooms, numberOfOccupiedRooms int, schedulerName string, managementApiClient *framework.APIClient) {
+	schedulerInfoRequest := &maestroApiV1.GetSchedulersInfoRequest{}
+	schedulerInfoResponse := &maestroApiV1.GetSchedulersInfoResponse{}
+	require.Eventually(t, func() bool {
+		err := managementApiClient.Do("GET", fmt.Sprintf("/schedulers/info?game=%s", autoscalingTestGame), schedulerInfoRequest, schedulerInfoResponse)
+		require.NoError(t, err)
+		require.NotEmpty(t, schedulerInfoResponse.GetSchedulers())
+		schedulerInfo := getSchedulerInfo(schedulerName, schedulerInfoResponse.GetSchedulers())
+		if schedulerInfo.GetAutoscaling() == nil {
+			return false
+		}
+
+		return schedulerInfo.GetRoomsReady() == int32(numberOfReadyRooms) &&
+			schedulerInfo.GetRoomsOccupied() == int32(numberOfOccupiedRooms)
+	}, time.Minute*3, time.Second*1, "Timeout waiting for scheduler to reach expected state")
```

Comment:
Executing in my host the test was broken here, what do you think to print the desired number of rooms for occupied and ready?